### PR TITLE
Do not ICE with a precision flag in formatting str and no format arguments

### LINKED ISSUE #66065: fmt::Display compiler panic: len is 0 but the index is 0
Hi there! This is the second time I've compiled code with Rust ever, and my first time filing a bug report, so I may need some guidance if the info here isn't quite enough to work with :). 

Learning about println! in Rust by Example, I incorrectly entered the below code. I looked at the documentation and realized I had made a mistake, but the compiler still panicked and recommended I open a bug report.  I really believe in Rust and want to see this project succeed, hopefully this information is helpful!

- I tried this code:  

println!("Pi is roughly: {:.*}, 3, 3.141592")

- Expected to see this happen:

"Pi is roughly 3.142"

- Instead, this happened:

thread 'rustc' panicked at 'index out of bounds: the len is 0 but the index is 0', /rustc/625451e376bb2e5283fc4741caa0a3e8a2ca4d54/src/libcore/slice/mod.rs:2715:10
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.

error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports

note: rustc 1.38.0 (625451e37 2019-09-23) running on x86_64-unknown-linux-gnu

## Meta

`rustc --version --verbose`:rustc 1.38.0 (625451e37 2019-09-23)
binary: rustc
commit-hash: 625451e376bb2e5283fc4741caa0a3e8a2ca4d54
commit-date: 2019-09-23
host: x86_64-unknown-linux-gnu
release: 1.38.0
LLVM version: 9.0

Backtrace:

thread 'rustc' panicked at 'index out of bounds: the len is 0 but the index is 0', /rustc/625451e376bb2e5283fc4741caa0a3e8a2ca4d54/src/libcore/slice/mod.rs:2715:10
stack backtrace:
   0: backtrace::backtrace::libunwind::trace
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.34/src/backtrace/libunwind.rs:88
   1: backtrace::backtrace::trace_unsynchronized
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.34/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:47
   3: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:36
   4: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:200
   5: std::panicking::default_hook
             at src/libstd/panicking.rs:214
   6: rustc::util::common::panic_hook
   7: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:481
   8: std::panicking::continue_panic_fmt
             at src/libstd/panicking.rs:384
   9: rust_begin_unwind
             at src/libstd/panicking.rs:311
  10: core::panicking::panic_fmt
             at src/libcore/panicking.rs:85
  11: core::panicking::panic_bounds_check
             at src/libcore/panicking.rs:61
  12: syntax_ext::format::expand_preparsed_format_args
  13: syntax_ext::format::expand_format_args_impl
  14: <F as syntax::ext::base::TTMacroExpander>::expand
  15: syntax::ext::expand::MacroExpander::expand_fragment
  16: syntax::ext::expand::MacroExpander::expand_crate
  17: rustc_interface::passes::configure_and_expand_inner::{{closure}}
  18: rustc::util::common::time
  19: rustc_interface::passes::configure_and_expand_inner
  20: rustc_interface::passes::configure_and_expand::{{closure}}
  21: rustc_data_structures::box_region::PinnedGenerator<I,A,R>::new
  22: rustc_interface::passes::configure_and_expand
  23: rustc_interface::queries::Query<T>::compute
  24: rustc_interface::queries::<impl rustc_interface::interface::Compiler>::expansion
  25: rustc_interface::interface::run_compiler_in_existing_thread_pool
  26: std::thread::local::LocalKey<T>::with
  27: syntax::with_globals
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
query stack during panic:
end of query stack

### PR DESCRIPTION
**Pull request by estebank**
Fix #66065.

### CODE REVIEW

**Comment by estebank** [comment]
@rust-lang/compiler sorry for the wide scatter approach, but this is a tiny PR that would fix a stable ICE and would like to merge it and backport before 1.39 leaves the station (although I understand if declined).

**Comment by jonas-schievink** [comment]
The 1.39 beta->stable promotion has already happened earlier today, so I'm not sure how likely this is to happen. cc @rust-lang/release

**Comment by estebank** [comment]
@jonas-schievink fair. Does that mean that 1.40-beta has already been cut? In that case it can follow regular backporting to 1.40-beta later in the week/next week.

This is not a critical ICE, it just felt to me way too easy to trigger.

**Comment by jonas-schievink** [comment]
AFAIK that will be done tomorrow (the master->beta promotion, that is)

**Comment by Mark-Simulacrum** [comment]
1.39 has been fully cut, any backports targeting beta in the release week need to be quite significant impact (e.g., ICE -> error fixes almost certainly don't qualify IMO). 1.40 beta has not yet been cut (that will happen [tomorrow](https://forge.rust-lang.org/release/process.html#promote-master-to-beta-t-2-days-tuesday)).

File: src/test/ui/issues/issue-66065.rs [hunk 1]
```diff
@@ -0,0 +1,3 @@
+fn main() {
```
Comment (Centril) [inline src/test/ui/issues/issue-66065.rs]: Can you give the test a more informative file name + add a comment describing what was wrong?

Comment (estebank) [reply src/test/ui/issues/issue-66065.rs]: Moved it with all the other formatting tests.

**Comment by pnkfelix** [comment]
Discussed in T-compiler meeting. beta-accepted, and stable-accepted (with understanding that this is not very high priority to fix and that release team is trusted to make the call with respect to when/whether to issue a point release).

**Pull request merged**